src/v/rpc/service.h

```diff
@@ -92,9 +92,12 @@ struct service::execution_helper {
                            */
                           vassert(
                             effective_version == version,
-                            "Unexpected encoding at effective {} != {}",
+                            "Unexpected encoding at effective {} != {}. Input "
+                            "{} Output {}",
                             effective_version,
-                            version);
+                            version,
+                            typeid(Input).name(),
```

Comment:
nit, we have `type_str<T>()` for this purpose. just realized that we have RTTI enabled and are using `dynamic_cast<>`, but probably we should to minimize the usage of it or even eliminate it if possible.